include/SFML/Graphics/SpriteBatch.hpp

```diff
@@ -0,0 +1,251 @@
+    ///                 rendererd. This value is used only when
+    ///                 BatchMode::DepthSort is used.
+    ///
+    ////////////////////////////////////////////////////////////
+    void draw(const Batchable& drawable, float depth = 0.0f);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Batches an array of vertices
+    ///
+    /// \param vertices  The array of vertices to be batched
+    /// \param count     The vertex count to batch
+    /// \param type      The primitive type that the vertices form
+    /// \param texture   The texture to use for rendering
+    /// \param transform The transform to apply on the vertices
+    /// \param depth     The depth at which the drawable will be
+    ///                  rendererd. This value is used only when
+    ///                  BatchMode::DepthSort is used.
+    ///
+    ////////////////////////////////////////////////////////////
+    void draw(const Vertex*    vertices,
+              std::size_t      count,
+              PrimitiveType    type,
+              const Texture*   texture,
+              const Transform& transform = Transform::Identity,
+              float            depth     = 0.0f);
+
+private:
+    ////////////////////////////////////////////////////////////
+    /// \brief Pushes a triangle into internal storage
+    ///
+    /// \param a         First vertex
+    /// \param b         Second vertex
+    /// \param c         Third vertex
+    /// \param transform Transform to apply before inserting
+    /// \param texture   Texture to apply to triangle
+    /// \param depth     The depth to use for depth sorting
+    ///
+    ////////////////////////////////////////////////////////////
+    void pushTriangle(const Vertex&    a,
+                      const Vertex&    b,
+                      const Vertex&    c,
+                      const Transform& transform,
+                      const Texture*   texture,
+                      float            depth);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Renders the given vertices
+    ///
+    /// \param vertices    The array of vertices to render
+    /// \param vertexCount The number of vertices in the array
+    /// \param texture     The texture to use
+    ///
+    ////////////////////////////////////////////////////////////
+    void sendVertices(const Vertex* vertices, std::size_t vertexCount, const Texture* texture);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Holds information about a batched triangle
+    ///
+    ////////////////////////////////////////////////////////////
+    struct TriangleInfo
+    {
+        const Texture* texture = nullptr;
+        float          depth   = 0.0f;
+
+        TriangleInfo() = default;
+        TriangleInfo(const Texture* theTexture, float theDepth) : texture(theTexture), depth(theDepth)
+        {
+        }
```

Comment:
I believe we can omit these constructors if we want to rely on aggregate initialization instead.

Reply:
> I believe we can omit these constructors if we want to rely on aggregate initialization instead.

Do we want that, though? The rest of SFML has "proper" constructors. Having dedicated functions also allows to do custom logic in the future (e.g. validations).

Reply:
I'm not sure if this simple data struct will ever need special logic, though.